### DISCUSSION TITLE
Support Apollo Federation

### DISCUSSION THREAD

**Discussion by benhaynes** [directus/directus#3012] [Feature Requests]
Needs research and an assessment of how "far off" we are currently, as well as testing against the 80/20 rule.

**Comment by valernyx** [comment]
+1

This is a deal-breaker for us if we cannot integrate Directus' GQL API with our federated graph setup.  I can't really find much on this other than this discussion post!  Are there any plans to include this in the near-term (or is it already supported)?

**Reply by rijkvanzanten** [reply]
Nope! No official timelines yet. You're the first person to bring it up in a long while I'm afraid, so this hasn't been on the list of priorities. That being said, if you'd like to expedite this, it would help out a lot if there is a clear list of things that need to be done in order to support Apollo Federation properly, or it would help out even more if you'd like to open a PR with those changes 🙂

**Reply by valernyx** [reply]
> Nope! No official timelines yet. You're the first person to bring it up in a long while I'm afraid, so this hasn't been on the list of priorities. That being said, if you'd like to expedite this, it would help out a lot if there is a clear list of things that need to be done in order to support Apollo Federation properly, or it would help out even more if you'd like to open a PR with those changes 🙂

Thank you for the quick response, much appreciated!  That's a bummer, because I do love everything else I've seen/researched so far with regard to Directus!  If I make progress on where the gaps are for federation support, I'll be sure to circle back around 👍

**Reply by rijkvanzanten** [reply]
It seems like there's a handful of resolvers we need to add to the GraphQL service (sounds like a doable task): https://www.apollographql.com/docs/federation/federation-spec/#federation-schema-specification but I'm not sure what else is required in order to make it integrate smoothly

**Comment by shadiramadan** [comment]
This is critical if you are managing existing GraphQL APIs.
It grants the ability to merge/extend schemas and have parts handled by different services as necessary.

Otherwise it is a headache to manage multiple GraphQL endpoints.
I don't think this is necessary for the system schemas but definitely so for the models.

We'd need the ability to define certain types / fields as shareable, external, etc.

This support along with properly marking required variables! https://github.com/directus/directus/issues/10610 really need to happen.

I keep running into these frustrating edge cases that are making me rethink integration.  Directus is SO CLOSE in so many aspects but keeps falling short when it comes to feature completion!

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
